aspnet/fundamentals/localization.rst

```diff
@@ -0,0 +1,337 @@
+
+- ``QueryStringRequestCultureProvider`` attempts to determine the culture based on the query string.
+- ``CookieRequestCultureProvider`` attempts to determine the culture based on a client-side cookie.
+- ``AcceptLanguageHeaderRequestCultureProvider`` attempts to determine the culture based on the Accept-Language value in the request header.
+
+When you instantiate a ``RequestLocalizationOptions`` object, the constructor instantiates all three of the aforementioned culture providers and adds them to the ``RequestLocalizationOptions.RequestCultureProviders`` collection.
+
+The order in which these providers is added to the collection is important as the first provider to return a non-null value from its implemented ``IRequestCultureProvider.DetermineRequestCulture`` method "wins". For example, let's say the URI includes a query string that specifies Spanish as the culture, but the Accept-Language value in the HTTP request header is set to English. Assuming the provider order above, the ``QueryStringRequestCultureProvider`` would return the culture for Spanish, so the ``AcceptLanguageHeaderRequestCultureProvider`` would not get a chance to evaluate the request.
+
+To remove, add to, or change the ordering of the providers used - including specifying your own custom ``IRequestCultureProvider`` implementation - you simply modify the ``RequestLocalizationOptions.RequestCultureProviders`` member to suit your app's needs. In the following example, the code removes all of the providers from a ``RequestLocalizationOptions`` object and adds the single provider it wants used (``QueryStringRequestcultureProvider``).
+
+.. code-block:: c#
+	:emphasize-lines: 6-9
+
+	public void Configure(IApplicationBuilder app, IStringLocalizer<Startup> SR)
+	{
+		// Add the platform handler to the request pipeline.
+		app.UseIISPlatformHandler();
+
+		var options = new RequestLocalizationOptions();
+		options.RequestCultureProviders.Clear();
+		options.RequestCultureProviders.Add(new QueryStringRequestCultureProvider());
+		app.UseRequestLocalization(options);
+
+		...
+	}
+
+Using the QueryStringRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When you instantiate a ``RequestLocalizationOptions`` object, the ``QueryStringRequestCultureProvider`` is added to the list of providers as the first provider to be used in attempting to determine the culture. The ``QueryStringRequestCultureProvider`` looks for the existence of two keys in the query string: "culture" and "ui-culture". If the provider finds a value for "culture", but not for "ui-culture", the UI culture is set to the culture value. Likewise, if the provider finds a value for "ui-culture", but not for culture, the culture is set to the UI culture value. Therefore, when using the ``QueryStringRequestCultureProvider``, all of the following URIs will result in a culture and UI culture of Spanish.
```

Comment:
> will result in a culture and UI culture of Spanish

I believe the [full display name](https://msdn.microsoft.com/en-us/library/ee825488(v=cs.20).aspx) of "es-ES" is actually "Spanish - Spain". There are multiple cultures that speak Spanish.